src/main/java/org/spongepowered/api/conversation/Conversation.java

```diff
@@ -0,0 +1,242 @@
+     */
+    void removeConversant(Conversant conversant);
+
+    /**
+     * Removes every {@link Conversant} from the conversation.
+     */
+    void removeAllConversants();
+
+    /**
+     * If the specified {@link Conversant} is in the conversation it changes their
+     * {@link ExternalChatHandler} to the one you specify. If they are not in the
+     * conversation, instead use {@link #addConversant(Conversant)}
+     *
+     * @param conversant The conversant which you want to modify
+     * @param externalChatHandler The external chat handler for the conversant
+     */
+    void setChatHandler(Conversant conversant, ExternalChatHandler externalChatHandler);
+
+    /**
+     * Gets the chat handler which handles incoming messages for a
+     * {@link Conversant} if they are in this conversation.
+     *
+     * @param conversant The conversant to get the handler for
+     * @return The handler of the specified {@link Conversant}, if present
+     */
+    Optional<ExternalChatHandler> getChatHandler(Conversant conversant);
+
+    /**
+     * Whether or not this conversation has ended.
+     *
+     * @return Whether or not this conversation has ended
+     */
+    boolean ended();
+
+    /**
+     * Sets whether or not {@link Conversant} output should be caught.
+     *
+     * @param catches If {@link Conversant} output should be caught
+     */
+    void catchOutput(boolean catches);
+
+    /**
+     * Gets if {@link Conversant} output should be caught.
+     *
+     * @return Whether or not {@link Conversant} output should be caught
+     */
+    boolean catchesOutput();
+
+    /**
+     * Sets whether or not {@link Conversant}s can use commands.
+     *
+     * @param allow If {@link Conversant}s can use commands
+     */
+    void allowCommands(boolean allow);
+
+    /**
+     * Gets if {@link Conversant}s can use commands.
+     *
+     * @return Whether or not {@link Conversant}s can use commands
+     */
+    boolean allowsCommands();
+
+    /**
+     * Gets every ending handler added to the conversation. Generally there is
+     * only one.
+     *
+     * @return All ending handlers for the conversation
+     */
+    Set<EndingHandler> getEndingHandlers();
+
+    /**
+     * Adds an ending handler which is processed when the conversation ends.
+     *
+     * @param endingHandler The ending handler to add
+     */
+    void addEndingHandler(EndingHandler endingHandler);
+
+    /**
+     * Removes the specified ending handler, so it is never processed.
+     *
+     * @param endingHandler The ending handler to removed
+     * @return Whether or not the specified ending handler was removed
+     */
+    boolean removeEndingHandler(EndingHandler endingHandler);
+
+    /**
+     * A convenience method to get the id of the conversation archetype.
+     *
+     * @return The id of the conversation archetype
+     */
+    default String getId() {
+        return getArchetype().getId();
```

Comment:
Well I would do it the other wa round. Remove the `this.` from method calls. You only need it for field access.